### ISSUE TITLE
Snowflake: Found unparsable section 'notify'

### ISSUE THREAD

**Issue by peter-malcolm-bw** [sqlfluff/sqlfluff#5638]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Sqlfluff throws a parsing error when we try and refer to a column named 'notify' in Snowflake. I think this restriction added as  part of #5272.

I've checked and notify isn't a reserved keyword in Snowflake: https://docs.snowflake.com/en/sql-reference/reserved-keywords. Although it is defined as a keyword in this file: https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_snowflake_keywords.py#L57

### Expected Behaviour

This script would parse without errors. i.e. notify would be an accepted keyword. 

### Observed Behaviour

Given the script: 

```sql
select notify
```

SQLFluff raises a parsing error:

```shell
== [test.sql] FAIL
L:   1 | P:   8 |  PRS | Line 1, Position 8: Found unparsable section: 'notify'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
```

### How to reproduce

Given the script `test.sql`: 

```sql
select notify
```

Running: ```sqlfluff lint test.sql```

SQLFluff raises the parsing error below:

```shell
== [test.sql] FAIL
L:   1 | P:   8 |  PRS | Line 1, Position 8: Found unparsable section: 'notify'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
```

### Dialect

Snowflake

### Version

sqlfluff==2.3.5 
python==3.9.12

### Configuration

Default config.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
